src/api/routes/rosetta/construction.ts

```diff
@@ -297,5 +324,161 @@ export function createRosettaConstructionRouter(db: DataStore): RouterWithAsync
+      res.status(400).json(RosettaErrors.emptyPublicKey);
+      return;
+    }
+
+    const recipientAddress = options.token_transfer_recipient_address;
+    if (!recipientAddress) {
+      res.status(400).json(RosettaErrors.invalidRecipient);
+      return;
+    }
+    const senderAddress = options.sender_address;
+
+    if (!senderAddress) {
+      res.status(400).json(RosettaErrors.invalidSender);
+      return;
+    }
+
+    const accountInfo = await new StacksCoreRpcClient().getAccount(senderAddress);
+
+    let tokenTransferOptions: UnsignedTokenTransferOptions | UnsignedMultiSigTokenTransferOptions;
+
+    if (publicKeys.length !== 1) {
+      //TODO support multi-sig in the future.
+      res.status(400).json(RosettaErrors.needOnePublicKey);
+      return;
+    } else {
+      if (publicKeys[0].curve_type !== 'secp256k1') {
+        res.status(400).json(RosettaErrors.invalidCurveType);
+        return;
+      }
+      // signel signature
+      tokenTransferOptions = {
+        recipient: recipientAddress,
+        amount: new BN(amount),
+        fee: new BN(fees),
+        publicKey: publicKeys[0].hex_bytes,
+        network: GetStacksTestnetNetwork(),
+        nonce: accountInfo.nonce ? new BN(accountInfo.nonce) : new BN(0),
+      };
+    }
+
+    const transaction = await makeUnsignedSTXTokenTransfer(tokenTransferOptions);
+    const unsignedTransaction = transaction.serialize();
+    const hexBytes = digestSha512_256(unsignedTransaction).toString('hex');
+    const response: RosettaConstructionPayloadResponse = {
+      unsigned_transaction: unsignedTransaction.toString('hex'),
+      payloads: [
+        {
+          address: senderAddress,
+          hex_bytes: '0x' + hexBytes,
+          signature_type: 'ecdsa',
+        },
+      ],
+    };
+    res.json(response);
+  });
+
+  //construction/combine endpoint
+  router.postAsync('/combine', async (req, res) => {
+    const valid: ValidSchema = await rosettaValidateRequest(req.originalUrl, req.body);
+    if (!valid.valid) {
+      res.status(400).json(makeRosettaError(valid));
+      return;
+    }
+
+    const combineRequest: RosettaConstructionCombineRequest = req.body;
+    const signatures = combineRequest.signatures;
+
+    if (has0xPrefix(combineRequest.unsigned_transaction)) {
+      res.status(400).json(RosettaErrors.invalidTransactionString);
+      return;
+    }
+
+    if (signatures.length === 0) {
+      res.status(400).json(RosettaErrors.noSignatures);
+      return;
+    }
+
+    let unsigned_transaction_buffer: Buffer;
+    let transaction: StacksTransaction;
+
+    try {
+      unsigned_transaction_buffer = hexToBuffer('0x' + combineRequest.unsigned_transaction);
+      transaction = deserializeTransaction(BufferReader.fromBuffer(unsigned_transaction_buffer));
+    } catch (e) {
+      res.status(400).json(RosettaErrors.invalidTransactionString);
+      return;
+    }
+
+    if (signatures.length !== 1) res.status(400).json(RosettaErrors.needOnlyOneSignature);
```

Comment:
Same question here -- is this what other Rosetta implementations do?

Reply:
Yes!
 https://github.com/iotexproject/iotex-core-rosetta-gateway/blob/master/services/construction.go#L65

Reply:
Alright, we should create a github issue for tracking multi-sig support once the Rosetta spec adds it

Reply:
Created issue https://github.com/blockstack/stacks-blockchain-api/issues/289